src/contracts/DatumRegistry.ts

```diff
@@ -0,0 +1,67 @@
+import { Contract, ethers } from 'ethers';
+
+import { IDatumCreation } from '../types/IDatumCreation';
+import { datumRegistryAbi } from './abi';
+import { LeafKeyCoder } from '../index';
+
+export class DatumRegistryContract {
+  contract: Contract;
+
+  constructor(providerOrSigner: ethers.providers.Provider | ethers.Signer, datumRegistryAddress: string) {
```

Comment:
I would suggest to rename this param to signerOrProvider. It is the same as ethers documentation https://docs.ethers.io/v5/api/contract/contract/

Reply:
Nice catch.